### PR TITLE
Fix AK1001 code fix for lambda expression with no block body

### LINKED ISSUE #55: AK1001 - giving wrong code fix and warning? when using with ReceiveAsync?
**Version Information**
Version of Akka.NET? 1.5.15
Which Akka.NET Modules? Akka.Analyzers

**Describe the bug**
When using Akka.Analyzers I now see the following warning (which in our solution turns into an error because we threat warnings as errors).

image

The code fix however suggests the following 'fix'. But outside the ReceiveAsync (in the constructor of my actor in this case) I don't have a valid Sender. So this code fix is wrong IMO (please explain if I'm wrong).

image

**Expected behavior**
I expect that this is the correct way to fix it:
image

**Additional question** is (if I need to create a separate issue for this, please let me know), why is this wrong?
image
As I've been using this over the past years on a lot of places and it always was working fine. We however always await tasks in our actors and never do async work in our actors which we don't await.

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #55

## Changes

- Check to see if the original invocation expression was inside a lambda expression with no block body
- Convert the lamda expression body to a block before inserting the new sender var

Converts
```csharp
Receive<string>(str => MessageHandler(str).PipeTo(Sender));
```
to
```csharp
Receive<string>(str =>
{
    var sender = this.Sender;
    MessageHandler(str).PipeTo(sender);
});
```

### CODE REVIEW

File: src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs [hunk 1]
```diff
@@ -60,16 +60,6 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
-        // Find an appropriate insertion point for the local variable
-        var insertionPoint = invocationExpr.FirstAncestorOrSelf<StatementSyntax>();
-
-        if (insertionPoint == null)
-            // Unable to find a valid insertion point
-            return document;
-
-        // Insert the local variable declaration at the found insertion point
-        editor.InsertBefore(insertionPoint, senderVariable);
-
```
Comment (Arkatufus) [inline src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs:72]: Need to reshuffle this code to the end

File: src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs [hunk 2]
```diff
@@ -93,8 +83,40 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            var originalBody = SyntaxFactory.ExpressionStatement(newExprSyntax);
+            
+            // Insert the local variable declaration at the start of the lambda block
+            var bodyBlock = SyntaxFactory.Block(
+                senderVariable.WithTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed), 
+                originalBody.WithTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed));
+            
+            // Replace the original with the new one
+            var newLambdaExpr = lambdaExpr.WithBody(null).WithBlock(bodyBlock);
+            editor.ReplaceNode(lambdaExpr, newLambdaExpr);
```
Comment (Arkatufus) [inline src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs:103]: Insert a block body if the original invocation was not enclosed inside a code block

**Review by Arkatufus — COMMENTED** [review]
Self review

File: src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/Akka.Analyzers.Fixes/AK1000/MustCloseOverSenderWhenUsingPipeToFixer.cs:103]: Great catch

**Review by Aaronontheweb — APPROVED** [review]
LGTM - approve

**Pull request merged**